apps/portal/app/.client/privy-refresh.tsx

```diff
@@ -1,22 +1,60 @@
-import { useEffect } from 'react'
+import { useCallback, useEffect, useState } from 'react'
 
+import LoadingLogo from '@components/loading-logo'
 import { usePrivy } from '@privy-io/react-auth'
-import { useRevalidator } from '@remix-run/react'
+import { useLocation, useNavigate, useRevalidator } from '@remix-run/react'
 
-export default function PrivyRefresh() {
+export default function PrivyRefresh({
+  refreshPath,
+  redirectTo,
+}: {
+  refreshPath: string
+  redirectTo: string
+}) {
   const { ready, getAccessToken } = usePrivy()
-  const { revalidate } = useRevalidator()
+  const [accessToken, setAccessToken] = useState<string | null>(null)
 
-  useEffect(() => {
-    async function refresh() {
-      if (ready) {
-        await getAccessToken()
-        revalidate()
-      }
+  const refresh = useCallback(async () => {
+    try {
+      if (!ready) return
+
+      console.log('Getting access token...')
+      const idToken = await getAccessToken()
+      console.log('Access token:', idToken)
+      setAccessToken(idToken)
+    } catch (error) {
+      console.error('Failed to refresh session:', error)
     }
+  }, [ready])
 
+  useEffect(() => {
     refresh()
-  }, [ready, revalidate])
+  }, [refresh])
+  useEffect(() => {
+    if (accessToken) {
+      // instead of revalidating, redirect to same route and replace true
+      console.log('Redirecting to', `${refreshPath}?redirectTo=${redirectTo}`)
```

Comment:
Good idea - this should also resolve some of the redirect questions being raised